### PR TITLE
Convert motion controller to Twist and Pose

### PR DESCRIPTION
**Pull request by abrandemuehl**
ROS usually uses Twist and Pose messages for controllers. I've updated all of the motion controller code to use these types, as well as the nodes and libraries that build on top of the motion controller.

Unfortunately, this is a really large pull request, because I also made changes to the helpers that make them easier to use (zero value constructors, for example) and the changes are very tightly coupled with the changes to the motion controller.

### CODE REVIEW

**Comment by abrandemuehl** [comment]
@shubhankar12 Review?

File: src/motion/motion_controller.cpp [hunk 1]
```diff
@@ -1,71 +1,50 @@
...
-  if (drpy.droll != 0.0 || drpy.dpitch != 0.0 || drpy.dyaw != 0.0) {
-    // Set a timeout for 1 second
-    timer = nh_.createTimer(ros::Duration(1.0),
-                            &MotionController::VelocityTimeout, this, true);
-  }
-}
-
-void MotionController::SetXYZ(const robosub::LinearPosition xyz) {
-  setpoint_xyz_ = xyz;
+void MotionController::SetPos(const geometry_msgs::PoseStamped pos) {
```
Comment (shubhankar12) [inline src/motion/motion_controller.cpp]: This should be `SetPose`. SetPos is misleading.

File: src/motion/rcmavros.cpp [hunk 2]
```diff
@@ -14,33 +18,41 @@ const std::string kModeDepthHold = "ALT_HOLD";
...
 
   // Service clients for arming and changing the mode of the sub
-  arming_client_ = nh_.serviceClient<CommandBool>("/mavros/cmd/arming");
-  mode_client_ = nh_.serviceClient<SetMode>("/mavros/set_mode");
+  arming_client_ = nh_.serviceClient<CommandBool>(kArmingService);
+  mode_client_ = nh_.serviceClient<SetMode>(kModeService);
 
   SetMode mode_cmd;
   mode_cmd.request.base_mode = 0;
   mode_cmd.request.custom_mode = kModeDepthHold;
```
Comment (shubhankar12) [inline src/motion/rcmavros.cpp]: Set the mode from the launch file.

File: src/experimental/barrelroll.cpp [hunk 3]
```diff
@@ -2,32 +2,34 @@
 #include <ros/ros.h>
 #include <cmath>
 #include <string>
-#include "helpers/AngularVelocity.h"
+
```
Comment (shubhankar12) [inline src/experimental/barrelroll.cpp]: I thought this file had comments. Add comments. Especially the while loop in main.

File: src/motion/rcmavros.cpp (hunk 2 above)
Comment (abrandemuehl) [reply src/motion/rcmavros.cpp]: The motion controller controls the mode. ArduSub responds to commands differently based on the mode, so it needs to be the only one in control of the mode.

**Pull request merged**